src/nfa.rs

```diff
@@ -90,7 +90,7 @@ impl<'r, 't> Nfa<'r, 't> {
                 //    soon as the last thread dies.
                 if matched
                    || (!at.is_beginning() && self.prog.anchored_begin) {
-                    break;
+                    return true;
```

Comment:
Does this pass tests? You can't just return `true` here because there might not be a match.